### PR TITLE
Feat/purgatory and punchlist

### PR DESCRIPTION
**Pull request by queenieho**
Includes:
- payment/review
- style payment/complete
- navbar updates properly as application progresses
- update in move-in/choose-date
- update in application when answer for pets changes
- bug fix in BAM
- update date picker in background check information
- logout actually logs out

### CODE REVIEW

File: src/cljs/apply/sections/personal/income.cljs [hunk 1]
```diff
@@ -45,8 +38,9 @@
 (defmethod db/step-complete? step
```
Comment (allenjprice) [inline src/cljs/apply/sections/personal/income.cljs]: ensure that loading spinners are fired/torn down for `::upload-income-verification!` event

File: src/cljs/apply/sections/payment/review.cljs [hunk 2]
```diff
@@ -35,26 +38,219 @@
 
 (defmethod db/step-complete? step
```
Comment (allenjprice) [inline src/cljs/apply/sections/payment/review.cljs]: the "finish and pay" button is _visually_ disabled until the TOS agreement is checked, but clicking the button when it should be disabled still works.

**Comment by allenjprice** [comment]
One more thing - Can we modify the purgatory view to only show sections that the applicant has completed?

If I reload the application before, say, selecting a move-in date, i get a purgatory view that looks like the image below, where i see things like "invalid date".

image

File: src/cljs/apply/sections/payment/review.cljs (hunk 2 above)
Comment (queenieho) [reply src/cljs/apply/sections/payment/review.cljs]: this was a flaw in my button component, it is fixed now

File: src/cljs/apply/sections/personal/background_check_info.cljs [hunk 3]
```diff
@@ -74,29 +74,56 @@
 ;; views ========================================================================
 
 
+(defn- is-leap?
+  [year]
+  (cond (zero? (mod year 400)) true
```
Comment (allenjprice) [inline src/cljs/apply/sections/personal/background_check_info.cljs]: Prefer `rem` to `mod here - see [the docs for `rem`](https://clojuredocs.org/clojure.core/rem) for an explainer

**Pull request merged**